### PR TITLE
add elements for waiting room to car

### PR DESCRIPTION
**Pull request by ditmirDVSA**
mes-7023

### CODE REVIEW

File: src/support/globalelements/globalelements.json [hunk 1]
```diff
@@ -118,7 +118,41 @@
...
+        "eyesight-pass-btn":"#eyesight-pass",
+        "eyesight-fail-btn":"#eyesight-fail",
+        "eyesight-validation-error-msg":"#waiting-room-to-car-eyesight-validation-text",
+        "tell-me-container":"#tell-me-card",
+        "tell-me-selector":"#tell-me-selector",
+        "tell-me-questions-options-modal":"div[role='radiogroup']",
+        "tell-me-questions-options":".select-interface-option",
+        "tell-me-validation-error-msg":"#waiting-room-to-car-tell-me-validation-text",
+        "modal-cancel-btn":"//button[contains(text(), 'Cancel')]",
+        "modal-submit-btn":"//button[contains(text(), 'Submit')]",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:137]: I have tried to see if this one is in the elements tab when the simulator is open but it does not appear.

File: src/support/globalelements/globalelements.json [hunk 2]
```diff
@@ -118,7 +118,41 @@
...
+        "eyesight-test-container":"#eyesight-test-card",
+        "eyesight-pass-btn":"#eyesight-pass",
+        "eyesight-fail-btn":"#eyesight-fail",
+        "eyesight-validation-error-msg":"#waiting-room-to-car-eyesight-validation-text",
+        "tell-me-container":"#tell-me-card",
+        "tell-me-selector":"#tell-me-selector",
+        "tell-me-questions-options-modal":"div[role='radiogroup']",
+        "tell-me-questions-options":".select-interface-option",
+        "tell-me-validation-error-msg":"#waiting-room-to-car-tell-me-validation-text",
+        "modal-cancel-btn":"//button[contains(text(), 'Cancel')]",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:136]: I have tried to see if this one is in the elements tab when the simulator is open but it does not appear.

File: src/support/globalelements/globalelements.json [hunk 3]
```diff
@@ -118,7 +118,41 @@
...
+        {
+        "eyesight-test-container":"#eyesight-test-card",
+        "eyesight-pass-btn":"#eyesight-pass",
+        "eyesight-fail-btn":"#eyesight-fail",
+        "eyesight-validation-error-msg":"#waiting-room-to-car-eyesight-validation-text",
+        "tell-me-container":"#tell-me-card",
+        "tell-me-selector":"#tell-me-selector",
+        "tell-me-questions-options-modal":"div[role='radiogroup']",
+        "tell-me-questions-options":".select-interface-option",
+        "tell-me-validation-error-msg":"#waiting-room-to-car-tell-me-validation-text",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:132]: THere is a second validation error when the user selects a tell me questions but does not answer it. Please add it

File: src/support/globalelements/globalelements.json [hunk 4]
```diff
@@ -118,7 +118,41 @@
...
+        "modal-cancel-btn":"//button[contains(text(), 'Cancel')]",
+        "modal-submit-btn":"//button[contains(text(), 'Submit')]",
+        "vehicle-registration-container":"#registration-card",
+        "vehicle-registration-input":"#registration",
+        "vehicle-registration-validation-error-msg":"#waiting-room-to-car-registration-validation-text",
+        "transmission-type-validation-error-msg":"#transmission-validation-text",
+        "transmission-type-container":"#transmission-card",
+        "manual-transmission-btn":"#transmission-manual-a6f6e3c1-72b6-4d60-9315-b539031fe6c3",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:143]: The manual button when searched for does not exist

File: src/support/globalelements/globalelements.json [hunk 5]
```diff
@@ -118,7 +118,41 @@
...
+        "automatic-transmission-btn":"#transmission-automatic-a6f6e3c1-72b6-4d60-9315-b539031fe6c3",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:144]: The automatic button when searched for does not exist

File: src/support/globalelements/globalelements.json [hunk 6]
```diff
@@ -118,7 +118,41 @@
...
+        "instructor-registration-input":"#instructor-registration",
+        "accompanied-by-container":"#accompaniment-card",
+        "accompanied-instructor-option":"#accompaniment-Instructor",
+        "accompanied-supervisor-option":"#accompaniment-Supervisor",
+        "accompanied-interpreter-option":"#accompaniment-Interpreter",
+        "accompanied-other-option":"#accompaniment-Other",
+        "vehicle-details-container":"vehicle-details",
+        "vehicle-details-school-car-option":"#vehicle-details-school-car",
+        "vehicle-details-dual-option":"#vehicle-details-dual-control",
+        "view-test-centre-journal-btn":"//ion-button[contains(text(), 'View test centre journal')]",
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:155]: I have tried to see if this one is in the elements tab when the simulator is open but it does not appear.

File: src/support/globalelements/globalelements.json [hunk 7]
```diff
@@ -118,7 +118,41 @@
...
+        "continue-to-test-report-btn":"//ion-button[contains(text(), 'Continue to test report')]"
```
Comment (turnerc1) [inline src/support/globalelements/globalelements.json:156]: I have tried to see if this one is in the elements tab when the simulator is open but it does not appear.

**Review by turnerc1 — CHANGES_REQUESTED** [review]

File: src/support/globalelements/globalelements.json (hunk 1 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:137]: Thats using path to find a button by name as it didn't have an ID

File: src/support/globalelements/globalelements.json (hunk 5 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:144]: The ID will need to be changed then to remove the extra characters after #transmission-automatic

File: src/support/globalelements/globalelements.json (hunk 6 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:155]: these are xpaths

File: src/support/globalelements/globalelements.json (hunk 7 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:156]: again ... these are xpaths  no ID for them exists

File: src/support/globalelements/globalelements.json (hunk 2 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:136]: path .. no ID exists for them

File: src/support/globalelements/globalelements.json [hunk 8]
```diff
@@ -118,7 +118,41 @@
...
+    "des-waiting-room-to-car": [
+        {
+        "eyesight-test-container":"#eyesight-test-card",
+        "eyesight-pass-btn":"#eyesight-pass",
+        "eyesight-fail-btn":"#eyesight-fail",
+        "eyesight-validation-error-msg":"#waiting-room-to-car-eyesight-validation-text",
+        "tell-me-container":"#tell-me-card",
+        "tell-me-selector":"#tell-me-selector",
+        "tell-me-questions-options-modal":"div[role='radiogroup']",
+        "tell-me-questions-options":".select-interface-option",
```
[the remark this answers is not in the document]
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:131]: you'll have to create a custom function to select radio options. We don't want to hard code these options here as they can change

File: src/support/globalelements/globalelements.json (hunk 4 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:143]: if that ID is subject to change and extra stuff has been added to make it dynamic , we should ask the developers to remove the dynamic part

File: src/support/globalelements/globalelements.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:131]: handled by search team which is re-usable here

[the remark this answers is not in the document]
Comment (turnerc1) [reply src/support/globalelements/globalelements.json:131]: Will you input search-teams function to fix this? If not please do.

**Review by turnerc1 — CHANGES_REQUESTED** [review]

File: src/support/globalelements/globalelements.json (hunk 3 above)
Comment (turnerc1) [reply src/support/globalelements/globalelements.json:132]: No there is a second validation message that appears

File: src/support/globalelements/globalelements.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:131]: this is not the scope of this ticket as far as I'm aware .. we'll implement it once we need it

File: src/support/globalelements/globalelements.json (hunk 3 above)
Comment (ditmirDVSA) [reply src/support/globalelements/globalelements.json:132]: that is the only one that shows up. looks correct to me 

Screenshot 2021-08-12 at 10 48 01

**Review by turnerc1 — APPROVED** [review]

**Pull request merged**